ahn/training/_decoder.py

```diff
@@ -124,7 +232,7 @@ def collate_fn(batch):
     return DataLoader(
         dataset,
         batch_size=config["deepspeed"]["train_micro_batch_size_per_gpu"],
-        num_workers=os.cpu_count() // dist.get_world_size(),
+        num_workers=4,
```

Comment:
Why did you using static value(4) instead of follow by world size?

Reply:
Because of following warning:
```UserWarning: This DataLoader will create 12 worker processes in total. Our suggested max number of worker in current system is 4, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.```